### ISSUE TITLE
ImportError: OpenCV loader: missing configuration file: ['config.py']. Check OpenCV installation.

### ISSUE THREAD

**Issue by jordisnake** [Nuitka/Nuitka#1275]
I have installed OpenCV v4.1 from source. I have successfully created binary file using Nuitka. After that when I run the binary file it produces Import Error.

`ImportError: OpenCV loader: missing configuration file: ['config.py']. Check OpenCV installation.`

Compiled with the following command:

`python3 -m nuitka --standalone --plugin-enable=pylint-warnings --plugin-enable=numpy --plugin-enable=pkg-resources --remove-output  --python-flag=no_site main.py`

Platform: Jetson Nano running on Ubuntu 18.04 LTS
Python Version: 3.6
OpenCV 4.1.1 (compiled from source)
Nuitka 0.6.15 installed through PIP3

**Comment by kayhayen** [comment]
Do you have some sort of minimal example so I can reproduce this.

**Comment by savelyevno** [comment]
Had the same issue for cv2 ver. 4.5.4.58, resolved after downgrading to ver. 3.4.8.29

**Comment by AI-Ying** [comment]
This is due to the different management methods of Anaconda and PIP，If you are using an Anaconda environment, it is recommended to use Conda to install OpenCV.
`conda install -c https://conda.anaconda.org/menpo opencv`

**Comment by hashuser** [comment]
I have the same issue.

**Comment by hashuser** [comment]
`
Traceback (most recent call last):
  File "C:\Users\home\AppData\Local\Temp\ON5948~1\main.py", line 1, in <module>
  File "C:\Users\home\AppData\Local\Temp\ON5948~1\cv2\__init__.py", line 180, in <module cv2>
  File "C:\Users\home\AppData\Local\Temp\ON5948~1\cv2\__init__.py", line 110, in bootstrap
  File "C:\Users\home\AppData\Local\Temp\ON5948~1\cv2\__init__.py", line 108, in load_first_config
ImportError: OpenCV loader: missing configuration file: ['config.py']. Check OpenCV installation.
`

**Comment by hashuser** [comment]
**Command**
`
nuitka --mingw64  --onefile --standalone --show-progress --plugin-enable=numpy --output-dir=C:\Users\home\Desktop\Python\1 C:\Users\home\PycharmProjects\bamboo\main.py
`

**Comment by hashuser** [comment]
```python
import cv2
import numpy as np
from math import pow
import os

def calculate_length(x,x_size,y_size,actural_x_unit_size,actural_y_unit_size):
    if cv2.contourArea(x) < (x_size * y_size * 0.001):
        raise Exception
    top = tuple(x[x[:, :, 1].argmin()][0])
    bottom = tuple(x[x[:, :, 1].argmax()][0])
    left = tuple(x[x[:, :, 0].argmin()][0])
    right = tuple(x[x[:, :, 0].argmax()][0])
    length_tb = pow(pow(top[0] - bottom[0], 2) + pow(top[1] - bottom[1], 2), 0.5)
    actural_length_tb = pow(pow((top[0] - bottom[0]) * actural_x_unit_size, 2) + pow((top[1] - bottom[1]) * actural_y_unit_size, 2), 0.5)
    length_lr = pow(pow(left[0] - right[0], 2) + pow(left[1] - right[1], 2), 0.5)
    actural_length_lr = pow(pow((left[0] - right[0]) * actural_x_unit_size, 2) + pow((left[1] - right[1]) * actural_y_unit_size, 2), 0.5)
    if actural_length_tb >= actural_length_lr:
        return top, bottom, length_tb, actural_length_tb
    else:
        return left, right, length_lr, actural_length_lr

def get_mask(img):
    imgb = cv2.GaussianBlur(img, (9, 9,), 0)
    hsv = cv2.cvtColor(imgb, cv2.COLOR_BGR2HSV)
    low_green = np.array([70/2, 25/100*255, 25/100*255])
    high_green = np.array([200/2, 100/100*255, 100/100*255])
    mask = cv2.inRange(imgb, low_green, high_green)
    #kernal = np.ones((3, 3))
    #mask = cv2.erode(mask, kernal, 2)
    #mask = cv2.dilate(mask, kernal, 2)
    return mask

def measure(name):
    img = cv2.imread(name)
    result = []
    x_size = img.shape[1]
    y_size = img.shape[0]
    if x_size < y_size:
        actural_x_unit_size = 210 / x_size
        actural_y_unit_size = 297 / y_size
    else:
        actural_x_unit_size = 297 / x_size
        actural_y_unit_size = 216 / y_size
    mask = get_mask(img)
    contours, _ = cv2.findContours(mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE)
    for x in range(len(contours)):
        try:
            a, b, length, actural_length = calculate_length(contours[x],x_size,y_size,actural_x_unit_size,actural_y_unit_size)
            #if length / max(x_size, y_size) > 0.05 and length < min(x_size, y_size) * 0.95:
            if length / max(x_size, y_size) > 0.05:
                result.append((a, b, str(round(actural_length, 1))))
                #cv2.drawContours(img, contours, x, (0, 0, 255), -1)
        except Exception as e:
            pass
    for x in result:
        cv2.line(img, x[0], x[1], (0, 0, 255), 10)
        font = cv2.FONT_HERSHEY_PLAIN
        cv2.putText(img, x[2] + ' mm', x[1], font, 5, (0, 0, 255), 3)
    cv2.imwrite(os.path.dirname(name)+'/r' + os.path.basename(name), img)
    #cv2.imwrite('m' + name, mask)

if __name__ == '__main__':
    while True:
        name = input()
        measure(name)
```

**Comment by jordisnake** [comment]
I made this simple example code:
```
import cv2

video_path = '/demo-video.mp4'
print("::C:: Video Path " + video_path)
cap0 = cv2.VideoCapture(video_path)
```

then compiled with:
`python3 -m nuitka --standalone --plugin-enable=numpy --python-flag=no_site main.py`

And when executing with`./main.dist/main` is when I get the error:

```
Traceback (most recent call last):
  File "/home/datatransfer/Documentos/proves/compilacio/main.dist/main.py", line 1, in <module>
  File "/home/datatransfer/Documentos/proves/compilacio/main.dist/cv2/__init__.py", line 89, in <module cv2>
  File "/home/datatransfer/Documentos/proves/compilacio/main.dist/cv2/__init__.py", line 58, in bootstrap
  File "/home/datatransfer/Documentos/proves/compilacio/main.dist/cv2/__init__.py", line 56, in load_first_config
ImportError: OpenCV loader: missing configuration file: ['config.py']. Check OpenCV installation.
```

**Comment by AI-Ying** [comment]
> **Environment** System: Windows10 Python Version: 3.9.7 OpenCV Version: opencv-python 4.5.4.58 Nuitka Version: 0.6.17.6 **Command** `nuitka --mingw64 --onefile --standalone --show-progress --plugin-enable=numpy --output-dir=C:\Users\home\Desktop\Python\1 C:\Users\home\PycharmProjects\bamboo\main.py`

This is because the config.py file of new opencv is dynamically loaded. Therefore, it cannot be analyzed by Depends, so you need to add dependencies by yourself
`python3 -m nuitka --standalone --plugin-enable=numpy --include-data-file={your site-packages path}/cv2/config*.py=cv2/ --python-flag=no_site main.py`

**Comment by AI-Ying** [comment]
> I made this simple example code:
> 
> ```
> import cv2
> 
> video_path = '/demo-video.mp4'
> print("Video Path " + video_path)
> cap0 = cv2.VideoCapture(video_path)
> ```
> 
> then compiled with: `python3 -m nuitka --standalone --plugin-enable=numpy --python-flag=no_site main.py`
> 
> And when executing with`./main.dist/main` is when I get the error:
> 
> ```
> Traceback (most recent call last):
>   File "/home/datatransfer/Documentos/proves/compilacio/main.dist/main.py", line 1, in <module>
>   File "/home/datatransfer/Documentos/proves/compilacio/main.dist/cv2/__init__.py", line 89, in <module cv2>
>   File "/home/datatransfer/Documentos/proves/compilacio/main.dist/cv2/__init__.py", line 58, in bootstrap
>   File "/home/datatransfer/Documentos/proves/compilacio/main.dist/cv2/__init__.py", line 56, in load_first_config
> ImportError: OpenCV loader: missing configuration file: ['config.py']. Check OpenCV installation.
> ```

You can also try this https://github.com/Nuitka/Nuitka/issues/1275#issuecomment-983200547

**Comment by kayhayen** [comment]
I think `--include-module=cv2.config` would be more appropiate, not sure if `cv2` insists on it being a data file. For package data, there is also an option where filenames are not needed.

**Comment by AI-Ying** [comment]
> I think `--include-module=cv2.config` would be more appropiate, not sure if `cv2` insists on it being a data file. For package data, there is also an option where filenames are not needed.

Thanks, the following code is taken from `cv/__ini__.py`，I also tried `include-module` and `include-package-data`, but did not take effect.  I think , `config* .py` cannot be imported as modules due to be loaded by the absolute path.
```python
LOADER_DIR = os.path.dirname(os.path.abspath(os.path.realpath(__file__)))
def load_first_config(fnames, required=True):
    for fname in fnames:
        fpath = os.path.join(LOADER_DIR, fname)
        if not os.path.exists(fpath):
            if DEBUG: print('OpenCV loader: config not found, skip: {}'.format(fpath))
            continue
        if DEBUG: print('OpenCV loader: loading config: {}'.format(fpath))
        exec_file_wrapper(fpath, g_vars, l_vars)
        return True
    if required:
        raise ImportError('OpenCV loader: missing configuration file: {}. Check OpenCV installation.'.format(fnames))

load_first_config(['config.py'], True)
load_first_config([
    'config-{}.{}.py'.format(sys.version_info[0], sys.version_info[1]),
    'config-{}.py'.format(sys.version_info[0])
], True)
```

**Comment by kayhayen** [comment]
@AI-Ying you are right, we should add it to `data-files.yml` then, so it becomes automatic. Anyone try a PR with that?

**Comment by ductandev** [comment]
I am using opencv 4.2.0 it resolved my problem

**Comment by kayhayen** [comment]
@ductandev that's actually useful to know

**Comment by kayhayen** [comment]
It*s solved on factory branch and will be in the 0.8 release.
